lib/streamlit/widgets.py

```diff
@@ -73,8 +74,21 @@ def get_widget_value(self, widget_id: str) -> Optional[Any]:
 
         return getattr(wstate, value_type)
 
+    def get_previous_widget_value(self, widget_id: str) -> Optional[Any]:
+        """Return the previous value of a widget, or None if no value was set."""
```

Comment:
Probably could be DRY with `_get_widget_value`, but it's not a big deal for the prototype.

Reply:
I somehow missed that `_get_widget_value` is 90% of what the methods do. Will be trivial to clean up.